src/fatx_disk.c

```diff
@@ -0,0 +1,294 @@
+    return FATX_STATUS_SUCCESS;
+}
+
+/*
+ * Reformat a disk as FATX.
+ */
+int fatx_disk_format(struct fatx_fs *fs, char const *path, size_t sector_size, enum fatx_format format_type, size_t sectors_per_cluster)
+{
+    struct fatx_partition_map_entry const *pi;
+    size_t f_offset, f_size;
+
+    if (format_type == FATX_FORMAT_NONE)
+    {
+        return FATX_STATUS_SUCCESS;
+    }
+
+    fatx_info(fs, "Writing refurb info...\n");
+    if (fatx_disk_write_refurb_info(path, 0, 0))
+    {
+        return FATX_STATUS_ERROR;
+    }
+
+    for (int i = 0; i < FATX_RETAIL_PARTITION_COUNT; i++)
+    {
+        pi = &fatx_partition_map[i];
+
+        fatx_info(fs, "-------------------------------------------\n");
+        fatx_info(fs, "Formatting partition %d (%c drive) ...\n", i, pi->letter);
+
+        /*
+         * Xapi initialization validates that the cluster size of retail
+         * partitions is 16kb when a game begins loading.
+         *
+         * For this reason, it is imperative that we do not let users
+         * configure the cluster size on retail partitions or many games
+         * will not load. Adjusting sector sizes, however, is okay.
+         */
+        if (fatx_disk_format_partition(fs, path, pi->offset, pi->size, sector_size, FATX_RETAIL_CLUSTER_SIZE / sector_size))
+        {
+            fatx_error(fs, " - failed to format partition %d\n", i);
+            return FATX_STATUS_ERROR;
+        }
+    }
+
+    if (format_type == FATX_FORMAT_RETAIL)
+    {
+        return FATX_STATUS_SUCCESS;
+    }
+
+    else if (format_type == FATX_FORMAT_F_TAKES_ALL)
+    {
+        fatx_drive_to_offset_size('f', &f_offset, &f_size);
+
+        fatx_info(fs, "-------------------------------------------\n");
+        fatx_info(fs, "Formatting partition %d (%c drive) ...\n", 5, 'f');
+
+        if (fatx_disk_format_partition(fs, path, f_offset, f_size, sector_size, sectors_per_cluster))
+        {
+            fatx_error(fs, " - failed to format partition %d (f-takes-all)\n", 5);
+            return FATX_STATUS_ERROR;
+        }
+    }
+
+    return FATX_STATUS_SUCCESS;
+}
+
+/*
+ * Format partition.
+ */
+int fatx_disk_format_partition(struct fatx_fs *fs, char const *path, size_t offset, size_t size, size_t sector_size, size_t sectors_per_cluster)
+{
+    int retval;
+
+    if (fatx_open_device(fs, path, offset, size, sector_size, sectors_per_cluster))
+    {
+        return FATX_STATUS_ERROR;
+    }
+
+    if (fatx_write_superblock(fs))
+    {
+        retval = FATX_STATUS_ERROR;
+        goto cleanup;
+    }
+
+    if (fatx_init_fat(fs))
+    {
+        retval = FATX_STATUS_ERROR;
+        goto cleanup;
+    }
+
+    if (fatx_init_root(fs))
+    {
+        retval = FATX_STATUS_ERROR;
+        goto cleanup;
+    }
+
+    retval = FATX_STATUS_SUCCESS;
+
+cleanup:
+    fatx_close_device(fs);
+    return retval;
+}
+
+/*
+ * Read refurb sector.
+ */
+int fatx_disk_read_refurb_info(char const *path)
```

Comment:
This doesn't give caller the refurb info, but it's also not used anywhere so I think we can just remove it for now